src/UartSerial.cpp

```diff
@@ -27,6 +23,8 @@ UartSerial::~UartSerial()
 
 size_t UartSerial::available()
 {
+  if (_fd == -1) return 0;
+
   int numBytesAvailable;
   ioctl(_fd, FIONREAD, &numBytesAvailable);
   return numBytesAvailable;
```

Comment:
Not really sure what to do about the high baud rate without adding some notion to the serial implementation about what the capabilities are of the device it will be talking to. I feel like that could get gross quickly.